network/transports_test.go

```diff
@@ -68,35 +74,77 @@ func TestHTTPRoundTripper(t *testing.T) {
 	}
 }
 
+//goland:noinspection HttpUrlsUsage
+const HTTPPrefix = "http://"
+
 func TestDialWithBackoff(t *testing.T) {
-	// Nobody's listening on a random port. Usually.
-	c, err := DialWithBackOff(context.Background(), "tcp4", "127.0.0.1:41482")
-	if err == nil {
-		c.Close()
-		t.Error("Unexpected success dialing")
+	t.Parallel()
+	t.Run("ConnectionRefused", testDialWithBackoffConnectionRefused)
+	t.Run("Timeout", testDialWithBackoffTimeout)
+	t.Run("Success", testDialWithBackoffSuccess)
+}
+
+func testDialWithBackoffConnectionRefused(t *testing.T) {
+	port := findUnusedPortOrFail(t)
+	addr := fmt.Sprintf("127.0.0.1:%d", port)
+	c, err := DialWithBackOff(context.Background(), "tcp4", addr)
+	closeOrFail(t, c)
+	if !errors.Is(err, syscall.ECONNREFUSED) {
+		t.Errorf("Unexpected error: %+v", err)
 	}
+}
 
+func testDialWithBackoffTimeout(t *testing.T) {
 	// Make the test short.
 	bo := backOffTemplate
-	bo.Steps = 2
+	bo.Steps = 1
+	bo.Duration = time.Millisecond
 
-	// Timeout. Use special testing IP address.
-	c, err = dialBackOffHelper(context.Background(), "tcp4", "198.18.0.254:8888", bo, sleepTO)
+	// Timeout. Use non-routable IP. See: https://stackoverflow.com/a/31581323/844449
```

Comment:
One of the [documentation IP ranges](https://datatracker.ietf.org/doc/html/rfc5737#section-3) (or a localhost IP like `127.200.100.0`) might make more sense than private IP space (and all are better than some real IP address belonging to Oracle, so thanks for fixing that!).

Reply:
Localhost's range isn't a good idea. We may hit, by chance, the port being open.